src/utilities/use-toggle.ts

```diff
@@ -0,0 +1,24 @@
+import {useState, useCallback} from 'react';
+
+export function useToggle(initialState: boolean) {
+  const [state, setState] = useState(initialState);
+  const toggle = useCallback(() => setState((state) => !state), []);
+
+  // cast needed to say this returns a two item array with the items in
+  // their specific positions instead of `(typeof state | typeof toggle)[]`
+  return [state, toggle] as [typeof state, typeof toggle];
+}
+
+export function useForcableToggle(initialState: boolean) {
```

Comment:
I think this is a bit of a stretch, I would just do this manually in the file itself and I don't think it comes up often enough to warrant a custom hook. Idk, feels a bit weird, if others think it's fine I'm fine with it though

Reply:
I've saw cases where we want to force set a boolean toggle to false (e.g. when you click a modal's close button / leave focus in popovers.

I think this crops up often enough to make a general purpose thing.